src/render/opengl/SDL_render_gl.c

```diff
@@ -889,7 +890,16 @@ static int GL_SetRenderTarget(SDL_Renderer *renderer, SDL_Texture *texture)
     /* Check FBO status */
     status = data->glCheckFramebufferStatusEXT(GL_FRAMEBUFFER_EXT);
     if (status != GL_FRAMEBUFFER_COMPLETE_EXT) {
-        return SDL_SetError("glFramebufferTexture2DEXT() failed");
+        if ((err = data->glGetError()) != GL_NO_ERROR) {
+            switch (err) {
+            case GL_INVALID_ENUM:
+                return SDL_SetError("glFramebufferTexture2DEXT() failed because of GL_INVALID_EMUM");
+            case GL_INVALID_OPERATION:
+                return SDL_SetError("glFramebufferTexture2DEXT() failed because of GL_INVALID_OPERATION");
+            }
+
+            return SDL_SetError("glFramebufferTexture2DEXT() failed with (err, status) = (%d, %d)", err, status);
```

Comment:
This `return` needs to be outside the `if`, or the function will report success if we have an incomplete framebuffer but glGetError() returns GL_NO_ERROR.